stream.go

```diff
@@ -225,8 +225,11 @@ func (s *Stream) sendWindowUpdate() error {
 			recvWindow = min(s.recvWindow*2, s.session.config.MaxStreamWindowSize)
 		}
 		if recvWindow > s.recvWindow {
-			s.recvWindow = recvWindow
-			_, delta = s.recvBuf.GrowTo(s.recvWindow, true)
+			status, err := s.session.memoryManager.ReserveMemory(int(delta))
+			if err == nil && status == MemoryStatusOK {
+				s.recvWindow = recvWindow
+				_, delta = s.recvBuf.GrowTo(s.recvWindow, true)
+			}
```

Comment:
wait here we have to release memory back if it is Caution or Critical but the error is nil, as it was actually reserved!